source/ns_cmdline.c

```diff
@@ -231,8 +240,14 @@ int history_command(int argc, char *argv[]);
 
 void default_cmd_response_out(const char *fmt, va_list ap)
 {
+#ifdef MBED_CONF_EXTERNAL_MBED_TRACE_ENABLED
     vprintf(fmt, ap);
     fflush(stdout);
+#else
+    log_disable_time_capture();
```

Comment:
this should be done outside of this library. 
See: https://github.com/ARMmbed/mbed-client-cli/blob/18018aaaa98b22d06f64e1704419e53f3b5f63e0/mbed-client-cli/ns_cmdline.h#L135. Most probably our test apps are using custom printing function so this change would not help. One reason why unified logging should be first optional so that teams can update test apps for this eg 🤔

Reply:
Well it’s up to the library owners to decide if default option they provide is fully compatible to mbed-os or not. Else you can add this to read me to show how it will be compatible to mbed-OS. Feature will be added to master first and will be part of minor release and not patch. That timing can be used to updated test apps.

Reply:
Original approach was that this library (as well as mbed-trace) is fully compatible with any OS which provide standard C libraries. All OS specific code is currently ouside of this library. It is okay to add some here but those need to be behind OS specific pre-compiler flags. I want to keep this in that way.

Reply:
Considering that we use this also with other platforms than Mbed OS (linux for. ex.), there is no reason to make the default Mbed OS.

Reply:
> Considering that we use this also with other platforms than Mbed OS (linux for. ex.), there is no reason to make the default Mbed OS.

Clarification over here, default is not Mbed OS. Default logging for Mbed OS will be unified logging and to make it compatible with existing client code, we have wrapper API's on top of unified logging which are same as mbed_trace.